features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/MemoryRequirementChecker.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.canReadSafe
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.readLinesSafe
+import java.io.File
+
+internal class MemoryRequirementChecker(
+    private val minRamSize: Int,
+    private val memInfoFile: File = File(MEM_INFO_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Long? = null
+
+    override fun checkMinimumRequirement(): Boolean {
+        return getMaxRAMSize() >= minRamSize
+    }
+
+    override fun name(): String = MEMORY_CHECK_NAME
+
+    override fun checkedValue(): Any? = checkedValue
+
+    private fun getMaxRAMSize(): Long {
+        if (!(memInfoFile.existsSafe(internalLogger) && memInfoFile.canReadSafe(internalLogger))) {
+            return 0L
+        }
+
+        val memorySizeKb = memInfoFile.readLinesSafe(internalLogger = internalLogger)?.mapNotNull { line ->
+            if (line.startsWith(MEM_TOTAL_REGEX)) {
+                val tokens = line.split(Regex("\\s+"))
+                if (tokens.size > 1) {
+                    try {
+                        tokens[1].toLong() // The memory value is in kB
```

Comment:
maybe `toLongOrNull` to avoid dealing with the exception? Or we need it in our telemetry?

Reply:
yes, if it can not be converted to Long, we probably want to know what happens in our telemetry.

Reply:
but what can be the action from our side in this case? not sure if this file can have a different format given that it is fixed https://man7.org/linux/man-pages/man5/proc_meminfo.5.html

Reply:
I intended to have the information about what exception that we can have when parsing the Long value, but probably we can't have any action even if we detect some errors, so let's remove the telemetry and change it to `toLongOrNull`